internal/api/shared/database/sql/query.go

```diff
@@ -122,3 +122,44 @@ func (d *DAO) buildQuery(query databaseModel.Query) (string, []interface{}, erro
 	}
 	return sqlQuery, args, nil
 }
+
+func generateProjectResourceDeleteQuery(tableName string, project string, name string) (string, []interface{}) {
```

Comment:
What about this to avoid some duplication ?

```go
func generateResourceDeleteQuery(tableName string, project string, name string) (string, []interface{}) {
        queryBuilder := sqlbuilder.NewDeleteBuilder().DeleteFrom(tableName)
	if len(name) > 0 {
		queryBuilder.Where(queryBuilder.Like(colName, fmt.Sprintf("%s%%", name)))
	}
	if len(project) > 0 {
		queryBuilder.Where(queryBuilder.Equal(colProject, project))
	}
	return queryBuilder.Build()
}

func (d *DAO) buildDeleteQuery(query databaseModel.Query) (string, []interface{}, error) {
	var sqlQuery string
	var args []interface{}
	switch qt := query.(type) {
	case *dashboard.Query:
		sqlQuery, args = generateResourceDeleteQuery(d.generateCompleteTableName(tableDashboard), qt.Project, qt.NamePrefix)
	case *datasource.Query:
		sqlQuery, args = generateResourceDeleteQuery(d.generateCompleteTableName(tableDatasource), qt.Project, qt.NamePrefix)
	case *folder.Query:
		sqlQuery, args = generateResourceDeleteQuery(d.generateCompleteTableName(tableFolder), qt.Project, qt.NamePrefix)
	case *globaldatasource.Query:
		sqlQuery, args = generateResourceDeleteQuery(d.generateCompleteTableName(tableGlobalDatasource), "", qt.NamePrefix)
	case *project.Query:
		sqlQuery, args = generateResourceDeleteQuery(d.generateCompleteTableName(tableProject), "", qt.NamePrefix)
	default:
		return "", nil, fmt.Errorf("this type of query '%T' is not managed", qt)
	}
	return sqlQuery, args, nil
}
```

Reply:
ah yes you are right. I don't what I was thinking🤦🏼